src/format.rs

```diff
@@ -0,0 +1,392 @@
+            if let Some(ver) = config.version {
+                return ver;
+            }
+        }
+        unreachable!("Default config didn't define version???");
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////
+//                                                                                //
+//                                                                                //
+//                                                                                //
+//                                audits.toml                                     //
+//                                                                                //
+//                                                                                //
+//                                                                                //
+////////////////////////////////////////////////////////////////////////////////////
+
+pub type AuditedDependencies = StableMap<String, Vec<AuditEntry>>;
+
+/// audits.toml
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct AuditsFile {
+    /// A map of criteria_name to details on that criteria.
+    pub criteria: StableMap<String, CriteriaEntry>,
+    /// Actual audits.
+    pub audits: AuditedDependencies,
+}
+
+impl AuditsFile {
+    pub fn validate(&self) -> Result<(), VetError> {
+        // TODO
+        // * check that each CriteriaEntry has 'description' or 'description_url'
+        // * check that all criteria are valid in:
+        //   * CriteriaEntry::implies
+        //   * AuditEntry::criteria
+        //   * DependencyCriteria
+        // * check that all 'audits' entries are well-formed
+        // * check that all package names are valid (with crates.io...?)
+        // * check that all reviews have a 'who' (currently an Option to stub it out)
+        // * catch no-op deltas?
+        Ok(())
+    }
+}
+
+/// Information on a Criteria
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
+pub struct CriteriaEntry {
+    /// Summary of how you evaluate something by this criteria.
+    pub description: Option<String>,
+    /// An alternative to description which locates the criteria text at a publicly-accessible URL.
+    /// This can be useful for sharing criteria descriptions across multiple repositories.
+    #[serde(rename = "description-url")]
+    pub description_url: Option<String>,
+    /// Criteria that this one implies
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    #[serde(default)]
+    pub implies: Vec<String>,
+}
+
+/// This is conceptually an enum
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct AuditEntry {
+    pub who: Option<String>,
+    pub notes: Option<String>,
+    pub criteria: String,
+    #[serde(flatten)]
+    pub kind: AuditKind,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+#[serde(untagged)]
+pub enum AuditKind {
+    Full {
+        version: Version,
```

Comment:
also the fact that this thing actually *works* is completely blowing my mind, i fucking love serde